source/common/router/router.cc

```diff
@@ -452,7 +475,14 @@ void Filter::onUpstreamReset(UpstreamResetType type,
       body = "upstream connect error or disconnect/reset before headers";
     }
 
-    chargeUpstreamCode(code, upstream_host);
+    chargeUpstreamCode(code, upstream_host,
+                       reset_reason.valid() &&
+                           reset_reason.value() == Http::StreamResetReason::Overflow);
+    // If we had non-5xx but still have been reset by backend or timeout, we
```

Comment:
This is only in the timeout_response_code_ case, right? Can you clarify comment?

Reply:
Not sure; can't we be reset by the backend without hitting a timeout, e.g. by having a TCP connection drop?

Reply:
Sorry yes, I was talking about looking at code below. I think code can only be non-5xx in the alt-timeout-response case.

Reply:
A few lines above, `code` is set to 503 in the non-timeout case.